debezium-core/src/main/java/io/debezium/transforms/UnwrapFromEnvelope.java

```diff
@@ -58,24 +59,61 @@
             .withDescription("Drop delete records converted to tombstones records if a processing connector "
                     + "cannot process them or a compaction is undesirable.");
 
+    private static final Field HANDLE_DELETES = Field.create("delete.handling.mode")
+            .withDisplayName("Handle delete records")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .withDefault(true)
+            .withDescription("How to handle delete records. Options are: "
+                    + "none - records are passed,"
+                    + "drop - records are removed,"
+                    + "rewrite - __delete field is added to records.");
+
     private boolean dropTombstones;
     private boolean dropDeletes;
-    private final ExtractField<R> delegate = new ExtractField.Value<R>();
+    private String handleDeletes;
+    private final ExtractField<R> afterDelegate = new ExtractField.Value<R>();
+    private final ExtractField<R> beforeDelegate = new ExtractField.Value<R>();
+    private final InsertField<R> removedDelegate = new InsertField.Value<R>();
+    private final InsertField<R> updatedDelegate = new InsertField.Value<R>();
 
     @Override
     public void configure(final Map<String, ?> configs) {
         final Configuration config = Configuration.from(configs);
-        final Field.Set configFields = Field.setOf(DROP_TOMBSTONES, DROP_DELETES);
+        final Field.Set configFields = Field.setOf(DROP_TOMBSTONES, HANDLE_DELETES);
         if (!config.validateAndRecord(configFields, logger::error)) {
             throw new ConnectException("Unable to validate config.");
         }
 
         dropTombstones = config.getBoolean(DROP_TOMBSTONES);
         dropDeletes = config.getBoolean(DROP_DELETES);
+        if (dropDeletes) {
+            handleDeletes = "drop";
+        } else {
+            handleDeletes = "none";
+        }
+        if (!config.getString(HANDLE_DELETES).isEmpty()) {
+            handleDeletes = config.getString(HANDLE_DELETES);
+        }
 
-        final Map<String, String> delegateConfig = new HashMap<>();
+        Map<String, String> delegateConfig = new HashMap<>();
+        delegateConfig.put("field", "before");
+        beforeDelegate.configure(delegateConfig);
+
+        delegateConfig = new HashMap<>();
         delegateConfig.put("field", "after");
-        delegate.configure(delegateConfig);
+        afterDelegate.configure(delegateConfig);
+
+        delegateConfig = new HashMap<>();
+        delegateConfig.put("static.field", "__deleted!");
```

Comment:
Please use constant fild. Is the exclamation mark in the name intentional?

Reply:
It is. Exclamation mark makes this field required in schema.